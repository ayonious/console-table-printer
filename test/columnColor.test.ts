import { Table } from '../index';

describe('Testing column colors', () => {
  it('should verify internal color states', () => {
    const p = new Table({
      columns: [
        { name: 'red_col', color: 'red' },
        { name: 'blue_col', color: 'blue' },
        { name: 'green_col', color: 'green' },
        { name: 'default_col' }
      ]
    });

    // Verify internal color states
    expect(p.table.columns[0].color).toBe('red');
    expect(p.table.columns[1].color).toBe('blue');
    expect(p.table.columns[2].color).toBe('green');
    expect(p.table.columns[3].color).toBeUndefined(); // default has no color

    p.addRow({
      red_col: 'red text',
      blue_col: 'blue text',
      green_col: 'green text',
      default_col: 'default text'
    });

    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  it('should handle color changes after column creation', () => {
    const p = new Table()
      .addColumn({ name: 'col1', color: 'red' })
      .addColumn('col2');
    
    // Change color after creation
    p.table.columns[0].color = 'blue';
    p.table.columns[1].color = 'green';

    expect(p.table.columns[0].color).toBe('blue');
    expect(p.table.columns[1].color).toBe('green');

    p.addRow({ col1: 'changed to blue', col2: 'changed to green' });
    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  it('should handle row and column color combinations', () => {
    const p = new Table({
      columns: [
        { name: 'red_col', color: 'red' },
        { name: 'blue_col', color: 'blue' }
      ]
    });

    p.addRows([
      { red_col: 'red column', blue_col: 'blue column' },
      { red_col: 'with green row', blue_col: 'with green row' }, { color: 'green' },
      { red_col: 'with yellow row', blue_col: 'with yellow row' }, { color: 'yellow' }
    ]);

    expect(p.table.columns[0].color).toBe('red');
    expect(p.table.columns[1].color).toBe('blue');

    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  it('should handle all available colors', () => {
    const p = new Table({
      columns: [
        { name: 'red', color: 'red' },
        { name: 'green', color: 'green' },
        { name: 'yellow', color: 'yellow' },
        { name: 'blue', color: 'blue' },
        { name: 'magenta', color: 'magenta' },
        { name: 'cyan', color: 'cyan' },
        { name: 'white', color: 'white' },
        { name: 'white_bold', color: 'white_bold' },
        { name: 'crimson', color: 'crimson' }
      ]
    });

    p.addRow({
      red: 'red text',
      green: 'green text',
      yellow: 'yellow text',
      blue: 'blue text',
      magenta: 'magenta text',
      cyan: 'cyan text',
      white: 'white text',
      white_bold: 'bold text',
      crimson: 'crimson text'
    });

    // Verify all colors are set correctly
    expect(p.table.columns[0].color).toBe('red');
    expect(p.table.columns[1].color).toBe('green');
    expect(p.table.columns[2].color).toBe('yellow');
    expect(p.table.columns[3].color).toBe('blue');
    expect(p.table.columns[4].color).toBe('magenta');
    expect(p.table.columns[5].color).toBe('cyan');
    expect(p.table.columns[6].color).toBe('white');
    expect(p.table.columns[7].color).toBe('white_bold');
    expect(p.table.columns[8].color).toBe('crimson');

    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  it('should handle color with unicode characters', () => {
    const p = new Table({
      columns: [
        { name: 'red_emoji', color: 'red' },
        { name: 'blue_emoji', color: 'blue' },
        { name: 'green_emoji', color: 'green' }
      ],
      charLength: {
        '❤️': 2,
        '💙': 2,
        '💚': 2
      }
    });

    p.addRows([
      { red_emoji: '❤️', blue_emoji: '💙', green_emoji: '💚' },
      { red_emoji: '❤️ text', blue_emoji: 'text 💙', green_emoji: '💚 colored' }
    ]);

    expect(p.table.columns[0].color).toBe('red');
    expect(p.table.columns[1].color).toBe('blue');
    expect(p.table.columns[2].color).toBe('green');

    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  it('should handle colors with different alignments and lengths', () => {
    const p = new Table({
      columns: [
        { name: 'left_red', color: 'red', alignment: 'left', maxLen: 15 },
        { name: 'center_blue', color: 'blue', alignment: 'center', maxLen: 15 },
        { name: 'right_green', color: 'green', alignment: 'right', maxLen: 15 }
      ]
    });

    const longText = 'This text is too long and will be truncated';
    p.addRows([
      { left_red: longText, center_blue: longText, right_green: longText },
      { left_red: 'short', center_blue: 'short', right_green: 'short' }
    ]);

    expect(p.table.columns[0].color).toBe('red');
    expect(p.table.columns[0].alignment).toBe('left');
    expect(p.table.columns[0].maxLen).toBe(15);

    expect(p.table.columns[1].color).toBe('blue');
    expect(p.table.columns[1].alignment).toBe('center');
    expect(p.table.columns[1].maxLen).toBe(15);

    expect(p.table.columns[2].color).toBe('green');
    expect(p.table.columns[2].alignment).toBe('right');
    expect(p.table.columns[2].maxLen).toBe(15);

    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  // Original test case preserved
  it('column colors are working', () => {
    // Create a table
    const p = new Table({
      columns: [
        { name: 'red_left_align_index', alignment: 'left', color: 'red' },
        { name: 'right_align_text', alignment: 'right' },
        { name: 'green_value', color: 'green' },
      ],
    });

    // add rows with color
    p.addRow(
      {
        red_left_align_index: 2,
        right_align_text: 'This row is blue',
        green_value: 10.212,
      },
      { color: 'blue' }
    );
    p.addRow(
      {
        red_left_align_index: 3,
        right_align_text: 'I would like some red wine please',
        green_value: 10.212,
      },
      { color: 'red' }
    );
    p.addRow(
      {
        red_left_align_index: 4,
        right_align_text: 'I would like some cyan wine please',
        green_value: 10.212,
      },
      { color: 'cyan' }
    );
    p.addRow(
      {
        red_left_align_index: 5,
        right_align_text: 'I would like some white_bold wine please',
        green_value: 10.212,
      },
      { color: 'white_bold' }
    );
    p.addRow(
      {
        red_left_align_index: 7,
        right_align_text: 'I would like some green gemuse please',
        green_value: 20.0,
      },
      { color: 'green' }
    );
    p.addRow(
      {
        red_left_align_index: 8,
        right_align_text: 'I would like some gelb bananen bitte',
        green_value: 100,
      },
      { color: 'yellow' }
    );

    // print
    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });
});
