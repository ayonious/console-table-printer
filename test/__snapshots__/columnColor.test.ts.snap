// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example: Print a simple Table with column colors column colors are working 1`] = `
"┌──────────────────────┬──────────────────────────────────────────┬─────────────┐
│[37m [0m[01mred_left_align_index[0m │[37m [0m[01m                        right_align_text[0m │[37m [0m[01mgreen_value[0m │
├──────────────────────┼──────────────────────────────────────────┼─────────────┤
│[37m [0m[31m2                   [0m │[37m [0m[34m                        This row is blue[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m3                   [0m │[37m [0m[31m       I would like some red wine please[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m4                   [0m │[37m [0m[36m      I would like some cyan wine please[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m5                   [0m │[37m [0m[01mI would like some white_bold wine please[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m7                   [0m │[37m [0m[32m   I would like some green gemuse please[0m │[37m [0m[32m         20[0m │
│[37m [0m[31m8                   [0m │[37m [0m[33m    I would like some gelb bananen bitte[0m │[37m [0m[32m        100[0m │
└──────────────────────┴──────────────────────────────────────────┴─────────────┘"
`;

exports[`Testing column colors column colors are working 1`] = `
"┌──────────────────────┬──────────────────────────────────────────┬─────────────┐
│[37m [0m[01mred_left_align_index[0m │[37m [0m[01m                        right_align_text[0m │[37m [0m[01mgreen_value[0m │
├──────────────────────┼──────────────────────────────────────────┼─────────────┤
│[37m [0m[31m2                   [0m │[37m [0m[34m                        This row is blue[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m3                   [0m │[37m [0m[31m       I would like some red wine please[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m4                   [0m │[37m [0m[36m      I would like some cyan wine please[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m5                   [0m │[37m [0m[01mI would like some white_bold wine please[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m7                   [0m │[37m [0m[32m   I would like some green gemuse please[0m │[37m [0m[32m         20[0m │
│[37m [0m[31m8                   [0m │[37m [0m[33m    I would like some gelb bananen bitte[0m │[37m [0m[32m        100[0m │
└──────────────────────┴──────────────────────────────────────────┴─────────────┘"
`;

exports[`Testing column colors should handle all available colors 1`] = `
"┌──────────┬────────────┬─────────────┬───────────┬──────────────┬───────────┬────────────┬────────────┬──────────────┐
│[37m [0m[01m     red[0m │[37m [0m[01m     green[0m │[37m [0m[01m     yellow[0m │[37m [0m[01m     blue[0m │[37m [0m[01m     magenta[0m │[37m [0m[01m     cyan[0m │[37m [0m[01m     white[0m │[37m [0m[01mwhite_bold[0m │[37m [0m[01m     crimson[0m │
├──────────┼────────────┼─────────────┼───────────┼──────────────┼───────────┼────────────┼────────────┼──────────────┤
│[37m [0m[31mred text[0m │[37m [0m[32mgreen text[0m │[37m [0m[33myellow text[0m │[37m [0m[34mblue text[0m │[37m [0m[35mmagenta text[0m │[37m [0m[36mcyan text[0m │[37m [0m[37mwhite text[0m │[37m [0m[01m bold text[0m │[37m [0mcrimson text │
└──────────┴────────────┴─────────────┴───────────┴──────────────┴───────────┴────────────┴────────────┴──────────────┘"
`;

exports[`Testing column colors should handle color changes after column creation 1`] = `
"┌─────────────────┬──────────────────┐
│[37m [0m[01m           col1[0m │[37m [0m[01m            col2[0m │
├─────────────────┼──────────────────┤
│[37m [0m[34mchanged to blue[0m │[37m [0m[32mchanged to green[0m │
└─────────────────┴──────────────────┘"
`;

exports[`Testing column colors should handle color with unicode characters 1`] = `
"┌───────────┬────────────┬─────────────┐
│[37m [0m[01mred_emoji[0m │[37m [0m[01mblue_emoji[0m │[37m [0m[01mgreen_emoji[0m │
├───────────┼────────────┼─────────────┤
│[37m [0m[31m       ❤️[0m │[37m [0m[34m        💙[0m │[37m [0m[32m         💚[0m │
│[37m [0m[31m  ❤️ text[0m │[37m [0m[34m   text 💙[0m │[37m [0m[32m 💚 colored[0m │
└───────────┴────────────┴─────────────┘"
`;

exports[`Testing column colors should handle colors with different alignments and lengths 1`] = `
"┌─────────────────┬─────────────────┬─────────────────┐
│[37m [0m[01mleft_red       [0m │[37m [0m[01m  center_blue  [0m │[37m [0m[01m    right_green[0m │
├─────────────────┼─────────────────┼─────────────────┤
│[37m [0m[31mThis text is   [0m │[37m [0m[34m This text is  [0m │[37m [0m[32m   This text is[0m │
│[37m [0m[31mtoo long and   [0m │[37m [0m[34m too long and  [0m │[37m [0m[32m   too long and[0m │
│[37m [0m[31mwill be        [0m │[37m [0m[34m    will be    [0m │[37m [0m[32m        will be[0m │
│[37m [0m[31mtruncated      [0m │[37m [0m[34m   truncated   [0m │[37m [0m[32m      truncated[0m │
│[37m [0m[31mshort          [0m │[37m [0m[34m     short     [0m │[37m [0m[32m          short[0m │
└─────────────────┴─────────────────┴─────────────────┘"
`;

exports[`Testing column colors should handle row and column color combinations 1`] = `
"┌─────────────────┬─────────────────┬────────┐
│[37m [0m[01m        red_col[0m │[37m [0m[01m       blue_col[0m │[37m [0m[01m color[0m │
├─────────────────┼─────────────────┼────────┤
│[37m [0m[31m     red column[0m │[37m [0m[34m    blue column[0m │[37m [0m[37m      [0m │
│[37m [0m[31m with green row[0m │[37m [0m[34m with green row[0m │[37m [0m[37m      [0m │
│[37m [0m[31m               [0m │[37m [0m[34m               [0m │[37m [0m[37m green[0m │
│[37m [0m[31mwith yellow row[0m │[37m [0m[34mwith yellow row[0m │[37m [0m[37m      [0m │
│[37m [0m[31m               [0m │[37m [0m[34m               [0m │[37m [0m[37myellow[0m │
└─────────────────┴─────────────────┴────────┘"
`;

exports[`Testing column colors should verify internal color states 1`] = `
"┌──────────┬───────────┬────────────┬──────────────┐
│[37m [0m[01m red_col[0m │[37m [0m[01m blue_col[0m │[37m [0m[01m green_col[0m │[37m [0m[01m default_col[0m │
├──────────┼───────────┼────────────┼──────────────┤
│[37m [0m[31mred text[0m │[37m [0m[34mblue text[0m │[37m [0m[32mgreen text[0m │[37m [0m[37mdefault text[0m │
└──────────┴───────────┴────────────┴──────────────┘"
`;
